pkg/health/checker/checker.go

```diff
@@ -213,75 +278,174 @@ func (c consulHealthChecker) WatchHealth(
 	}
 }
 
-func (c consulHealthChecker) WatchService(
+func (h healthChecker) WatchService(
 	serviceID string,
 	resultCh chan<- map[types.NodeName]health.Result,
 	errCh chan<- error,
 	quitCh <-chan struct{},
 	watchDelay time.Duration,
+	useHealthService bool,
+	status manifest.StatusStanza,
 ) {
 	defer close(resultCh)
-	var curIndex uint64 = 0
-
 	if watchDelay < time.Second {
 		watchDelay = time.Second
 	}
 
 	timer := time.NewTimer(0)
-	for {
-		select {
-		case <-quitCh:
-			return
-		case <-timer.C:
-			timer.Reset(watchDelay)
-			results, queryMeta, err := c.client.KV().List(consul.HealthPath(serviceID, "/"), &api.QueryOptions{
-				WaitIndex: curIndex,
-			})
-			if err != nil {
-				select {
-				case <-quitCh:
-					return
-				case errCh <- consulutil.NewKVError("list", consul.HealthPath(serviceID, "/"), err):
+	if !useHealthService {
```

Comment:
Is there any way to get this to send health check requests to both consul & the health service? I don't yet have enough confidence that the health service returns timely & correct health checks, which is why I want this code to send traffic to the trusted source (_consul_) and the new service so we can see how it behaves under production load levels.

Reply:
Would this always send to both or only when useHealthService is true?

Reply:
We can use the boolean to toggle whether or not this code sends to both.